app/policies/comment_policy.rb

```diff
@@ -0,0 +1,29 @@
+class CommentPolicy < ApplicationPolicy
+
+  def create?
+    @user.user?
+  end
+
+  def new?
+    @user.user?
+  end
+
+  def edit?
+    @user == comment.user
```

Comment:
can you please confirm if this is firing a query or not, if yes then we should compare on id perhaps?